packages/data/src/components/use-dispatch/use-dispatch.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Internal dependencies
+ */
+import useRegistry from '../registry-provider/use-registry';
+
+/**
+ * A custom react hook returning the current registry dispatch actions creators.
+ *
+ * Note: The component using this hook must be within the context of a
+ * RegistryProvider.
+ *
+ * @param {string} [storeName]  Optionally provide the name of the store from
+ *                             which to retrieve action creators. If not
+ *                             provided, the registry.dispatch function is
+ *                             returned instead.
+ *
+ * @example
+ * This illustrates a pattern where you may need to retrieve dynamic data from
+ * the server via the `useSelect` hook to use in combination with the dispatch
+ * action.
+ *
+ * ```jsx
+ * const { useDispatch, useSelect } = wp.data;
+ * const { useCallback } = wp.element;
+ *
+ * function Button( { onClick, children } ) {
+ *   return <button type="button" onClick={ onClick }>{ children }</button>
+ * }
+ *
+ * const SaleButton = ( { children } ) => {
+ *   const { stockNumber } = useSelect(
+ *     ( select ) => select( 'my-shop' ).getStockNumber()
+ *   );
+ *   const { startSale } = useDispatch( 'my-shop' );
+ *   const onClick = useCallback( () => {
+ *     const discountPercent = stockNumber > 50 ? 10: 20;
+ *     startSale( discountPercent );
+ *   }, [ stockNumber ] );
+ *   return <Button onClick={ onClick }>{ children }</Button>
+ * }
+ *
+ * // Rendered somewhere in the application:
+ * //
+ * // <SaleButton>Start Sale!</SaleButton>
+ * ```
+ * @return {Function}  A custom react hook.
+ */
+const useDispatch = ( storeName ) => {
+	const { dispatch } = useRegistry();
+	return storeName === void 0 ? dispatch : dispatch( storeName );
```

Comment:
First time I see `=== void 0`. I'd have done `!! storeName` personally. Curious to learn more here.

Reply:
ya `!! storeName` could have been used here.  Using `void 0` is just an old habit I sometimes slip into when I want to check against the `undefined` primitive value.  I don't know if it's still the case with modern browsers, but it was possible for the `undefined` global to be overwritten so it was just a protection.

Reply:
> I don't know if it's still the case with modern browsers, but it was possible for the `undefined` global to be overwritten so it was just a protection.

It's not been possible to override in any browser which conforms to ES5 and newer (IE9+).

https://www.ecma-international.org/ecma-262/5.1/#sec-15.1.1.3

A future maintainer most certainly will struggle to under what this code is doing. Not a big issue, but in general we should feel fine to compare against `undefined`.

Reply:
Babel might even take care of it too.

Reply:
I feel really old all of a sudden ;)

Reply:
> Babel might even take care of it too.

I was curious about this, and it doesn't seem like Babel does, but Uglify will convert it, yes.

Reply:
Interesting, maybe for the older targets?

Reply:
Likely because `void 0` is 3 fewer characters than `undefined` 😄

Reply:
> Likely because `void 0` is 3 fewer characters than undefined

Ha!  That's my new reason.